src/slack-model.jl

```diff
@@ -182,57 +202,113 @@ function NLPModels.objgrad!(nlp::SlackModels, x::AbstractVector, g::AbstractVect
   return f, g
 end
 
-function NLPModels.cons!(nlp::SlackModels, x::AbstractVector, c::AbstractVector)
+function NLPModels.cons_lin!(nlp::SlackModels, x::AbstractVector, c::AbstractVector)
+  @lencheck nlp.meta.nvar x
+  @lencheck nlp.meta.nlin c
+  n = nlp.model.meta.nvar
+  ns = nlp.meta.nvar - n
+  jlow, jupp, jrng = get_slack_lin(nlp)
+  nlow, nupp, nrng = length(jlow), length(jupp), length(jrng)
+  @views begin
+    cons_lin!(nlp.model, x[1:n], c)
+    I = n .+ jlow
+    c[jlow] -= x[I]
```

Comment:
Could we modify `jlow` directly here, or would it modify what's stored inside `meta`? Otherwise, a simply loop would be just as efficient and not incur allocations.

Reply:
Since the `jlow, jupp, jrng = get_slack_lin(nlp)` is used in all the functions in the linear API. Maybe, it is worth adding them to a `SlackNLPModelMeta`. What do you think ?
Making a loop will work as well, it will just complexity the code a lot I believe.

Reply:
good idea!

Reply:
I added these entries to the model directly for now. 
Adding, the new meta will be approx. 100 lines of codes https://github.com/JuliaSmoothOptimizers/NLPModels.jl/blob/main/src/nlp/meta.jl 
So, I suggest I do it in a separate PR.